### DISCUSSION TITLE
Problems for oblique electron firehose instability and high wave numbers.

### DISCUSSION THREAD

**Discussion by DustinSch** [danielver02/ALPS#177] [Q&A]
I'm encountering difficulties when trying to investigate electron firehose instabilities for oblique propagation angles. I'm using 'distribution_analyt' to generate the f0 tables. This method works well for proton firehose instabilities even at oblique propagation angles (see attached files PFHI).

However, numerical issues arise when dealing with electrons (see files EFHI attached). I think the problem is related to the momentum space resolution of the f0 tables, as you pointed out in your paper. I tried different resolutions in momentum space (and different P_max values) and k-space resolution.
My setup works for 0°, 20°, 40° but doesn’t work for 60°, when the magnitude of the wavenumber gets higher.

When I tried to reproduce the results from Li & Habaal (2000), which worked for the parallel propagation case, but not for the oblique ones (Li_EFHI). Adjusting the resolution in momentum space 'improves' the situation (higher resolution in the parallel direction keeps the real frequency at zero), but the resulting outcomes of the growth rates no longer align with the findings presented in Li and Habaal's paper. 

However, using the NHDS solver (using bimax) will lead to the correct results (Li_EFHI_NHDS).

I have tried finer and coarser momentum grids (between n_par = 800, n_perp=400 and n_par=100, n_perp = 50) and the same for the k-space. For the low wavenumbers at the beginning, the solutions seem to be ok, but as soon as I reach higher magnitudes of k, I encounter the shown issues (non-zero real frequency or wrong values for the growth rate).

Additionally, I receive an error message when the magnitude of k approaches the order of 30. It works fine until it says „Program received signal SIGBUS: Access to an undefined portion of a memory object" and ALPS stops then.

So I think the main problem is to determine the correct value for P_max and the corresponding resolution of the momentum space? 
Do you have any ideas how to find a working setup for different distribution functions?

Thanks in advance for any help!

[EFHI_0deg.pdf](https://github.com/danielver02/ALPS/files/14672086/EFHI_0deg.pdf)
[EFHI_40deg.pdf](https://github.com/danielver02/ALPS/files/14672087/EFHI_40deg.pdf)
[EFHI_60deg.pdf](https://github.com/danielver02/ALPS/files/14672088/EFHI_60deg.pdf)
[Habaal_efhi_24deg.pdf](https://github.com/danielver02/ALPS/files/14672089/Habaal_efhi_24deg.pdf)
[LiHabaal_efhi_0deg.pdf](https://github.com/danielver02/ALPS/files/14672090/LiHabaal_efhi_0deg.pdf)
[LiHabaal_efhi_60deg_nhds.pdf](https://github.com/danielver02/ALPS/files/14672091/LiHabaal_efhi_60deg_nhds.pdf)
[LiHabaal_efhi_60deg.pdf](https://github.com/danielver02/ALPS/files/14672092/LiHabaal_efhi_60deg.pdf)
[PFHI_75deg.pdf](https://github.com/danielver02/ALPS/files/14672093/PFHI_75deg.pdf)

**Comment by danielver02** [comment]
Hi Dustin,

Sorry for the delay with my response!

I have a few ideas that could be worth checking:

1) For the value of P_max, I would suggest using "a few" thermal momenta for P_max. In the case of an anisotropic distribution (like for your firehose instability), this means that you may want to use different P_max in the perpendicular and parallel directions. For a bi-Maxwellian, the ideal P_max is "a few" times sqrt(beta)*sqrt(ms/m_ref)*sqrt(n_ref/ns), where beta is the beta of the species (using Tperp for P_max in the perpendicular direction and Tpar for P_max in the parallel direction), ms is the mass of a particle of the species you're looking at, m_ref is the mass of a particle of your reference species (usually protons), n_ref is the number density of your reference species, and ns is the number density of the species you're looking at. For "a few", I would usually work with a value between 6 and 9. For kappa-distributions, you would need to go a bit higher in order to capture the tails. It's always good checking that the resolution of your f0-table is high enough to guarantee that it integrates to unity.

2) If the code struggles with the high perpendicular wavenumbers, you may want to run the code in "electron mode". This means that you would make the electrons your reference species (i.e., species number 1). The electrons then have mass, density, and charge of unity. The protons then get a mass of 1836 and a charge of minus one. All frequencies are then normalised in electron gyro-frequencies, and all wavenumbers are then normalised in inverse electron inertial lengths. It means that a different initial guess and scan profile are needed (and maybe some further numerical adjustments). However, this usually avoids errors due to the large k_perp (which I assume is the SIGBUS error that you get). If you want to play with electron mode a bit, NHDS can also do this, so that's probably an easier testbed to become familiar with this.

Please let me know if this works out for your problem. If not, you can also share your distribution input file here, and then I'll have a closer look.

Best wishes,
-Daniel

**Comment by kgklein** [comment]
For those following this thread along, it turns out that the velocity resolution can impact the damping rate, especially for non-propagating oblique solutions. We have started constructing a 'best practices' page to helping to resolve these kinds of issues: found [https://danielver02.github.io/ALPS/page/practices.html](here). Let us know if anyone using the code runs into these kinds of issues, and we will be glad to help.
